src/methods/defining_methods/hard/q9/MainTest.java

```diff
@@ -0,0 +1,63 @@
+package methods.defining_methods.hard.q9;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.utils.MethodUtil;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+
+public class MainTest extends BaseTest {
+  // Java
+  public Clause[] testSentence() {
+    TestOption.isInputTest = true;
+    TestOption.defaultInput = "3";
+    return new Clause[] {
+        new StringLiteral("Enter the number of courses you're taking this year:"),
+        new NewLine(),
+        new PlaceHolder(),
+    };
+  }
+
+  public void runMain() {
+    NumCoursesMethod.main(new String[0]);
+  }
+
+  static Stream<Arguments> coursesInputProvider() {
+    return Stream.of(
+        Arguments.of(3, 9),
+        Arguments.of(4, 12),
+        Arguments.of(50, 150),
+        Arguments.of(0, 0),
+        Arguments.of(10000, 30000));
+  }
+
+  @ParameterizedTest
+  @MethodSource("coursesInputProvider")
+  void methodCreditsCalcCorrect(int courses, int credits) throws Throwable {
+    String errorMsg = "Your method \"creditsCalc()\" does not correctly calculate the number of credits";
```

Comment:
Maybe add to this errorMsg just to make it more specific, like "... the number of credits the user will earn."